tests/contract-playground/src/TxOriginAccessControl.sol

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+contract OriginControlledTransfer {
+    address payable private _owner;
+
+    event TransferMade(address indexed sender, address indexed receiver, uint256 amount);
+
+    constructor() {
+        _owner = payable(msg.sender);
+    }
+
+    modifier onlyOwner() {
+        require(tx.origin == _owner, "Caller is not the owner");
```

Comment:
Should we try to capture custom error cases too? For example:

```solidity

if (tx.origin != _owner) {
    revert OwnerIsNotCaller();
}
```

Reply:
I'm adding a loop for the identifier of if (IfStatement), at the moment this part doesn't recognize the code you proposed to me:

For now, it correctly identifies the error in the require statements, but it doesn't detect it in the IfStatement. This is my code:

- https://github.com/scab24/aderyn/blob/29ac7fa4e8d18fc30a61785f0f3e9d25e0f8a2b8/aderyn_core/src/detect/low/tx_origin_access_control.rs#L51C1-L71

- https://github.com/scab24/aderyn/blob/%40secoalba%40/tests/contract-playground/src/TxOriginAccessControl.sol